package-res/resources/web/prompting/PromptPanel.js

```diff
@@ -569,54 +660,181 @@ define(['amd!cdf/lib/underscore', 'cdf/lib/Base', 'cdf/Logger', 'dojo/number', '
+            // we need to remove components from prompt panel component also
+            for (var i in toRemove) {
+              var toRemoveComponent = toRemove[i];
+              var index = panelComponent.components.indexOf(toRemoveComponent);
+              if (index > -1) {
+                panelComponent.components.splice(index, 1);
+              }
+            }
           }
+
+          _removeComponents.call(this, toRemove);
         },
 
+        /**
+         * Adds a set of components determined by the ParameterDefinitionDiffer#diff
+         *
+         * @name PromptPanel#_addComponentsByDiff
+         * @method
+         * @param {JSON} toAddDiff The group of parameters which need to be added
+         */
+        _addComponentsByDiff: function(toAddDiff) {
+          var panelComponent = this.dashboard.getComponentByName("prompt" + this.guid);
+
+          for (var groupName in toAddDiff) {
+            var addWrap = toAddDiff[groupName];
+            var params = addWrap.params;
+
+            var fieldComponents = [];
+            for (var i = 0; i < params.length; i++) {
+              var param = params[i];
+              var component = this._buildPanelForParameter(param); // returns a panel component
+              fieldComponents.push(component);
+            }
+
+            var groupPanel = this.dashboard.getComponentByName(groupName);
+            if (!groupPanel) {
+              groupPanel = _createWidgetForGroupPanel.call(this, addWrap.group, fieldComponents);
+              panelComponent.components.push(groupPanel);
+            } else {
+              groupPanel.components = groupPanel.components.concat(fieldComponents);
+            }
+          }
+
+          if (panelComponent.components.length > 0 && !_findSubmitComponent.call(this, panelComponent)) {
+            var submitPanel = _createWidgetForSubmitPanel.call(this);
+            panelComponent.components.push(submitPanel);
+          }
+
+          _addComponent.call(this, panelComponent);
+        },
+
+        /**
+         * Changes the data and selects the current value(s) of a set of components determined by the ParameterDefinitionDiffer#diff.
+         *
+         * @name PromptPanel#_changeComponentsByDiff
+         * @method
+         * @param {JSON} toChangeDiff The group of paramters which need to be have their data changed
+         */
+        _changeComponentsByDiff: function(toChangeDiff) {
+          for (var groupName in toChangeDiff) {
+            var changeWrap = toChangeDiff[groupName];
+            var params = changeWrap.params;
+
+            for (var i in params) {
+              var param = params[i];
+
+              // Find selected value in param values list and set it. This works, even if the data in valuesArray is different
+              var selectedValues = param.getSelectedValuesValue();
+              for (var j in selectedValues) {
+                this.setParameterValue(param, selectedValues[j]);
+              }
```

Comment:
Is it possible to have more than one selected value? I image that yes. Then, we need to set the list of values and not do this for the list, because in this case, only the last selected value will be the parameter value, which is wrong